### ISSUE TITLE
Note of things missing from API design discussion section (docs, pagination, throttling/limits, etc)

### ISSUE THREAD

**Issue by mlisanke** [jbelmont/api-workshop#7]
As mention at the meetup, it'd be great if there was some mention and detail given to API design aspects Not typically addressed. Some of these are,
1) Documentation (RESTful Introspection of the API through the REST API)
2) Pagination (handling data transfer limits using parameters to return secondary data pages)
3) Throttling/limits/QOS (API limits placed on user requests to prevent over user or metered service).
4) Security (Nounce/dynamic-paths used to restrict user access to a context)
5) other run-time API design (many run-time aspects like level 3 paths and advanced JSON data structure)

**Comment by jbelmont** [comment]
To address some of your points:

1. Mike can you clarify what you mean by RESTful Introspection.
2. There is a section for pagination in api design : https://github.com/jbelmont/api-workshop/blob/master/docs/api-design.md#pagination, I can add more about pagination concepts here but I believe this already does address some of your concern.
3. I can add a section like this in the gitbook in a TODO section.
 1. This blog post has good information on that  https://www.progress.com/blogs/throttling-database-using-rate-limits-for-sql-or-rest
4. This is also another section I can add in the TODO section.
5. I can bundle these into another section in API Design.

I adding a TODO section to the top README.md

**Comment by mlisanke** [comment]
have a look at https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwiX2_6x9pTgAhWuVN8KHQZ_D-8QFjAAegQIAxAB&url=https%3A%2F%2Fgithub.com%2Fvasilakisfil%2FIntrospected-REST&usg=AOvVaw0NZ37le8vs7jjBYGKvTWKG
I haven't worked on (more than tutorial) REST API but have heard a number of people speak about the intended purpose of introspection of REST API...it's equivalent to Introspection in CORBA, COM, DCOM as well as many programming language introspection models... allowing for full interpretation of an interface by its clients through the API itself. I googled and found the top link I added here, but; there were a number of links returned for 'rest api introspection' that can be reviewed. IMO, Any API which is embodied by a Interface implemented by protocol (rather than native code) should have Introspection implemented as a secondary requirement. Obviously, like HATEOAS, few interfaces get beyond practical requirements of client/server models of well-documented interfaces ... as programmers are adept at reading and coding to programming documentation. I think the future of the Internet is the Semantic Web and it's have Introspective REST API to obviate the need for such programming/coding Or AI's smarter than us will just use trial and error to guess how to crawl ancient human REST API :-p.
